File: CommonBase/src/main/java/com/common/base/callback/CallbackUI1.java
package com.common.base.callback;


import com.common.utils.UIHandler;

import androidx.annotation.NonNull;

/**
 * 1个参数，参数接口回调到UI线程
 *
 * @author LiuFeng
 * @data 2020/7/16 11:05
 */
public class CallbackUI1<T> implements CommonCallback1<T> {
    CommonCallback1<T> callback;

    public CallbackUI1(@NonNull CommonCallback1<T> callback) {
        this.callback = callback;
    }

    @Override
    public void onSuccess(T result) {
        if (callback != null) {
            UIHandler.run(() -> callback.onSuccess(result));
        }
    }

    @Override
    public void onError(int code, String desc) {
        if (callback != null) {
            UIHandler.run(() -> callback.onError(code, desc));
        }
    }
}

File: CommonBase/src/main/java/com/common/base/callback/CallbackUI1.kt
package com.common.base.callback

import com.common.utils.thread.UIHandler

/**
 * 1个参数，参数接口回调到UI线程
 *
 * @author LiuFeng
 * @data 2020/7/16 11:05
 */
class CallbackUI1<T>(val callback: CommonCallback1<T>) : CommonCallback1<T> {

    override fun onSuccess(result: T) {
        UIHandler.run { callback.onSuccess(result) }
    }

    override fun onError(code: Int, desc: String?) {
        UIHandler.run { callback.onError(code, desc) }
    }
}